tests/test_entrance_shims.py

```
@@ -0,0 +1,293 @@
+
+@pytest.mark.skipif(not DOOR.exists(), reason="setup door not landed yet")
+class TestCallGraph:
+    """F3: the call graph is strictly shim -> door -> engine."""
+
+    def test_shims_exec_the_door(self):
+        consumer_shim = REPO_ROOT / "scripts" / "local" / "bootstrap-consumer.sh"
+        for shim in (CREATE_PROJECT, BOOTSTRAP_MATERIALS, consumer_shim):
+            text = shim.read_text(encoding="utf-8")
+            assert (
+                "exec" in text and "bootstrap" in text
+            ), f"{shim.name} must exec the door"
```

Ruff (0.15.21)

[warning] Assertion should be broken down into multiple parts

(PT018)